hedera-node/src/main/java/com/hedera/services/utils/PlatformTxnRecord.java

```diff
@@ -0,0 +1,58 @@
+package com.hedera.services.utils;
+
+/*-
+ * ‌
+ * Hedera Services Node
+ * ​
+ * Copyright (C) 2018 - 2021 Hedera Hashgraph, LLC
+ * ​
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.swirlds.common.Transaction;
+
+import java.util.concurrent.TimeUnit;
+
+public class PlatformTxnRecord {
+
+    private Cache<Transaction, PlatformTxnAccessor> cache;
```

Comment:
Can you put some comments to explain what's the purpose of the `PlatformTxnRecord`? and also why a map of `<Transaction, PlatformTxnAccessor>`  is needed here?